pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -895,3 +965,73 @@ func statusError(logger klog.Logger, err error, kv ...interface{}) *framework.St
 	}
 	return framework.AsStatus(err)
 }
+
+func getAllocatedDeviceStatus(claim *resourceapi.ResourceClaim, deviceRequest *resourceapi.DeviceRequestAllocationResult) *resourceapi.AllocatedDeviceStatus {
+	for _, device := range claim.Status.Devices {
+		if deviceRequest.Device == device.Device && deviceRequest.Driver == device.Driver && deviceRequest.Pool == device.Pool {
+			return &device
+		}
+	}
+	return nil
+}
+
+// isClaimBound checks whether a given resource claim is successfully
+// bound to a device.
+func isClaimBound(logger klog.Logger, claim *resourceapi.ResourceClaim, pod *v1.Pod, nodeName string) (bool, error) {
+	for _, deviceRequest := range claim.Status.Allocation.Devices.Results {
+		if len(deviceRequest.BindingConditions) == 0 {
+			continue
+		}
+		deviceStatus := getAllocatedDeviceStatus(claim, &deviceRequest)
+		if deviceStatus == nil {
+			return false, nil
```

Comment:
if we remove the code above that creates the status (which I think we should), then this case is likely to hit for a while. we can add an logger as we do in the other cases, saying we are waiting for the driver to report status

In fact this should be an event, not just a log.

Reply:
>In fact this should be an event, not just a log.

What does this mean in terms of implementation?
For logs, I think I can use logger.Info etc., but for events, what would be appropriate to use?

Reply:
Yes, probably that's it. @pohly can say for sure.

Events are visible when the user does `kubectl describe` on the pod that's being scheduled. Log entries are not visible to users.

Reply:
Yes, that seems appropriate. There are two event APIs, we should use the same as the rest of the scheduler. See https://kubernetes.slack.com/archives/C20HH14P7/p1740748575144709